### PR TITLE
Bugfix/df 571 abroad jobads

### PR DESCRIPTION
**Pull request by cjpsycos**

### CODE REVIEW

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java [hunk 1]
```diff
@@ -456,7 +458,11 @@ private BoolQueryBuilder localityFilter(JobAdvertisementSearchRequest jobSearchR
             localityFilter.should(termsQuery(PATH_LOCATION_REGION_CODE, jobSearchRequest.getRegionCodes()));
         }
         if (isNotEmpty(jobSearchRequest.getCommunalCodes())) {
-            localityFilter.should(termsQuery(PATH_LOCATION_COMMUNAL_CODE, jobSearchRequest.getCommunalCodes()));
+            if (FILTER_COMMUNAL_CODE_ABROAD.equals(jobSearchRequest.getCommunalCodes())) {
```
Comment (pavax) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: we can submit multiple communcal-codes

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java [hunk 2]
```diff
@@ -88,6 +89,8 @@
     private static final String PATH_WORKLOAD_PERCENTAGE_MAX = PATH_CTX + "jobContent.employment.workloadPercentageMax";
     private static final String PATH_WORKLOAD_TIME_PERCENTAGE_MIN = PATH_CTX + "jobContent.employment.workloadPercentageMin";
     private static final String PATH_LANGUAGE_SKILL_CODE = PATH_CTX + "jobContent.languageSkills.languageIsoCode";
+    private static final String FILTER_COMMUNAL_CODE_ABROAD = "9999";
+    private static final String COUNTRY_ISO_CODE = "CH";
```
Comment (pavax) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: rename to SWITZERLAND

File: infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java [hunk 3]
```diff
@@ -456,6 +458,9 @@ private BoolQueryBuilder localityFilter(JobAdvertisementSearchRequest jobSearchR
             localityFilter.should(termsQuery(PATH_LOCATION_REGION_CODE, jobSearchRequest.getRegionCodes()));
         }
         if (isNotEmpty(jobSearchRequest.getCommunalCodes())) {
+            if (Arrays.asList(jobSearchRequest.getCommunalCodes()).contains(FILTER_COMMUNAL_CODE_ABROAD)) {
```
Comment (pavax) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: extract a method called: searchesForAboardJobs or something more meaningful

Comment (pavax) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: I'd create a unit in general for these search features -> if there are not unit tests yet please create a 'Technical' Debt Task with the high priority

Comment (birom) [inline infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: Please add a test to the [JobAdvertisementSearchControllerIntTest](https://github.com/alv-ch/job-ad-service/blob/4ea249e458/web/src/test/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/web/controller/JobAdvertisementSearchControllerIntTest.java).

Comment (pavax) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: DF-635

Comment (pavax) [reply infrastructure-elasticsearch/src/main/java/ch/admin/seco/jobs/services/jobadservice/infrastructure/elasticsearch/read/jobadvertisement/JobAdvertisementSearchService.java]: see: DF-635

**Pull request merged**